tests/test_polycos.py

```diff
@@ -0,0 +1,82 @@
+"""Test polycos."""
+
+import pytest
+import numpy as np
+from pint.polycos import Polycos
+from pint.models import get_model
+import pint.toa as toa
+from pinttestdata import datadir
+from pathlib import Path
+
+
+@pytest.fixture
+def polyco_file():
+    return Path(datadir) / "B1855_polyco.dat"
+
+
+@pytest.fixture
+def par_file():
+    return Path(datadir) / "B1855+09_polycos.par"
+
+
+def test_polycos_basic(polyco_file):
+    """Just run various features to make sure none of them throw errors.
+
+    Except the ones that should.
+    """
+    p = Polycos()
+    p.read_polyco_file(polyco_file)
+    table = p.polycoTable
+    entry = table["entry"][0]
+    print(entry)
+
+    # Test single float - arrays are tested later.
+    mjd = 55000.0
+    p.eval_spin_freq(mjd)
+    p.eval_abs_phase(mjd)
+    p.eval_phase(mjd)
+    p.find_entry(mjd)
+
+    with pytest.raises(ValueError):
+        p.find_entry(200000.0)
+
+
+def test_read_write_round_trip(tmpdir, polyco_file):
+    with open(polyco_file, "r") as f:
+        p1 = f.read()
+
+    output_polyco = tmpdir / "B1855_polyco_round_trip.dat"
+    p = Polycos()
+    p.read_polyco_file(polyco_file)
+    p.write_polyco_file(output_polyco)
+
+    with open(output_polyco, "r") as f:
+        p2 = f.read()
+
+    assert p1 == p2
+
+
+def test_generate_polycos(tmpdir, par_file):
```

Comment:
Personally, I don't think that adds much beyond the tests that already exist. All this module does is read/write in the polyco format - which is mostly tested by the couple of round-trip tests already in there.

That being said, I am not really well-versed in using `hypothesis` and am still learning how to work with `PINT` in general - so someone else might have to add those tests if needed.

Reply:
I'm less fussed about `hypothesis`, it seems overkill here, but I think it's valuable to check this works with a few non-default sets of parameters.

Reply:
I think I am misunderstanding what you mean. Can you provide an example of the kind of test you are suggesting here?

Reply:
Currently this test only generates polycos located at Arecibo, that are 300 minutes long, at 1400 MHz, et cetera. It's very possible that our polyco implementation breaks if asked to use, say, the barycenter, or longer or shorter segments, or... The easiest way to solve this is to use `@pytest.mark.parametrize` to run this test with several different combinations of parameters. Here is an example of doing this: https://github.com/nanograv/PINT/blob/4a0b6875483c5ceed1d8ea6228e535144a1f56fa/tests/test_pulse_number.py#L66

Reply:
Ah! I see. Sorry, I thought you were suggesting checking for correctness with these parameters. But if you just mean round-trip tests, then that should be easy.

Reply:
One could also check that the polyco prediction doesn't depend on the parameter settings, but since you're already checking against the model prediction that doesn't actually add anything.

(`hypothesis` could be used to thoroughly explore the space of possible combinations of inputs. But it's more complicated and you can spend a fair amount of time chasing non-problems.)

Reply:
Would you happen to know what obs_code tempo2 uses for barycenter/coe? I can't seem to figure this out.

Reply:
@ or bat

Reply:
For the geocenter I think it is 0 or coe

Reply:
Done. Thanks for the comments!